File: cadastroEscolarFront/src/app/models/quadroHorario.ts
import { Dia } from "./dia";
import { Horas } from "./horas";
import { Materia } from "./materia";
import { Turma } from "./turma";

export class QuadroHorario {
    id : number;

    turma : Turma;

    materia : Materia;

    horas : Horas;

    dia : Dia;

}

File: src/main/java/com/escola/cadastro/escolar/model/QuadroHorario.java
package com.escola.cadastro.escolar.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "QuadroHorario")
public class QuadroHorario implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    Turma turma;
    @ManyToOne
    Materia materia;

    @ManyToOne
    Horas horas;

    @ManyToOne
    Dia dia;

}
